benchexec/tools/ultimate.py

```diff
@@ -76,11 +76,12 @@ def __init__(self):
     @functools.lru_cache()
     def executable(self):
         exe = util.find_executable("Ultimate.py")
-        for (dirpath, dirnames, filenames) in os.walk(exe):
```

Comment:
I am not sure why this does not work for you in the first place. 

The ``None`` check seems necessary, but changing the parameter to ``os.path.dirname`` seems not.

Reply:
`exe` contains a path to a file, but `os.walk` needs a path to a directory, how can this work?

Reply:
os.walk takes a directory, not a file.  `exe` points to the file.

Reply:
I see, you are right. Good catch!